inference-engine/src/auto_plugin/auto_exec_network.cpp

```diff
@@ -0,0 +1,91 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <string>
+#include <vector>
+#include <memory>
+#include <map>
+#include <unordered_map>
+
+#include "ie_metric_helpers.hpp"
+#include "auto_exec_network.hpp"
+#include "auto_infer_request.hpp"
+
+namespace AutoPlugin {
+    using namespace InferenceEngine;
+
+AutoExecutableNetwork::AutoExecutableNetwork(const ExecutableNetwork& network,
+                                             const DeviceInformation& deviceInfo,
+                                             const bool               needPerfCounters) :
+    _deviceInfo(deviceInfo),
+    _network(network),
+    _config(deviceInfo.config.begin(), deviceInfo.config.end()),
+    _needPerfCounters(needPerfCounters) {
```

Comment:
let's call base class ctor with nullptr as executors.

Reply:
nullptr with base ctor will cause segment fault during test. maybe the default base ctor is a better way?

Reply:
this is because we also need to returns device inference request without wrapping with AutoInferRequest..

> GetAvailableDevices

Looks strange, because HETERO and MULTI does not support `AVAILABLE_DEVICES` metric. So, they should not be returned as available devices.